src/app/presenter/ChessBoardPresenter.ts

```diff
@@ -31,4 +33,13 @@ export class ChessBoardPresenter {
     const square: Square = { column: columns[position.x - 1], row: position.y as Row };
     return `${square.column.toLowerCase()}${square.row}`;
   }
+
+  private isPawnReadyForPromotion(position: Position): void {
+    const selectedSquare = this.translatePositionToAlgebraicNotation(position).toUpperCase();
+    const selectedSquareIsPawn = this.chessModel.squaresWithPiece[selectedSquare].name == 'Pawn';
```

Comment:
Zmiast 
`this.chessModel.squaresWithPiece[selectedSquare]`
Dobrze uzyc metody, ktora juz jest: `onPositionPiece`z ChessBoard.